tests/test_asset.py

```diff
@@ -306,3 +312,173 @@ def test_failure_bogus_uri():
+    )
+
+    # 1. Create a test user.
+    with tiledb.cloud._common.api_v4.ApiClient(configuration) as api_client:
+        api_instance = tiledb.cloud._common.api_v4.UsersApi(api_client)
+        request = (
+            tiledb.cloud._common.api_v4.models.user_create_request.UserCreateRequest(
+                display_name=DISPLAY_NAME,
+                email=EMAIL,
+                username=USERNAME,
+                password=PASSWORD,
+            )
+        )
+        user_create_response = api_instance.create_user(request)
+        assert user_create_response.data.display_name == DISPLAY_NAME
+        assert user_create_response.data.email == EMAIL
+        assert user_create_response.data.username == USERNAME
+
+        # For security, server API does not return the password.
+        # For testing only, we insert it into the creation response.
+        assert not hasattr(user_create_response.data, "password")
+        user_create_response.data._testing_password = PASSWORD
+        assert user_create_response.data._testing_password == PASSWORD
+
+    # 2. Validate the test user's email.
+    cxn = mysql.connector.connect(
+        host=MYSQL_HOST,
+        port=MYSQL_PORT,
+        database=MYSQL_DATABASE,
+        user=MYSQL_USER,
+        password=MYSQL_PASSWORD,
+    )
+    cursor = cxn.cursor()
+    cursor.execute("UPDATE users2 SET is_valid_email = 1 WHERE email = %s", (EMAIL,))
+    cxn.commit()
+    cursor.close()
+    cxn.close()
+
+    # 3. Authenticate and create a workspace.
+    with tiledb.cloud._common.api_v4.ApiClient(configuration) as api_client:
+        basic_payload = base64.b64encode(
+            f"{USERNAME}:{PASSWORD}".encode("ascii")
+        ).decode("ascii")
+        api_client.set_default_header(
+            "Authorization",
+            f"Basic {basic_payload}",
+        )
+        credential = models_v4.Credential(
+            aws=models_v4.AWSCredential(
+                access_key_id=AWS_ACCESS_KEY_ID, secret_access_key=AWS_SECRET_ACCESS_KEY
+            )
+        )
+        default_credential_request = models_v4.CredentialCreateRequest(
+            name="mydefaultcredential", provider="AWS", credential=credential
+        )
+        default_storage_setting_request = models_v4.StorageSettingsCreateRequest(
+            name="mydefaultstoragesetting",
+            is_default=False,
+            path=AWS_S3_PATH,
+            credentials_name="mydefaultcredential",
+        )
+        workspace_request = models_v4.WorkspaceCreateRequest(
+            name=WORKSPACE,
+            default_credential=default_credential_request,
+            default_storage_setting=default_storage_setting_request,
+        )
+        api_instance = tiledb.cloud._common.api_v4.WorkspacesApi(api_client)
+        workspace_create_response = api_instance.create_workspace(workspace_request)
+        assert workspace_create_response.data.name == WORKSPACE
+
+    return (user_create_response, workspace_create_response)
```

Comment:
Note: this fixture doesn't do any cleanup. I'm getting by without it, since the local server and dev stack I'm using are regularly restarted.